### PR TITLE
sync GitHub action distribution to S3

### PR DESCRIPTION
**Pull request by npalm**
To avoid we have to download the github action dist everytime on runner is created we cach the dist in S3.

### CODE REVIEW

File: modules/action-runner-binary-cache/lambdas/syncer/src/lambda.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { handle } from './syncer/handler';
+
+module.exports.handler = async (event: any, context: any, callback: any): Promise<any> => {
+  const statusCode = await handle();
```
Comment (gertjanmaas) [inline modules/action-runner-binary-cache/lambdas/syncer/src/lambda.ts:4]: StatusCode isn't necessary, this lambda isn't part of the API

File: modules/action-runner-binary-cache/lambdas/syncer/src/syncer/handler.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+interface ReleaseAsset {
+  name: string;
+  downloadUrl: string;
+}
+
+async function getLinuxReleaseAsset(): Promise<ReleaseAsset | undefined> {
+  const githubClient = new Octokit();
+  const linuxAssets = (
+    await githubClient.repos.getLatestRelease({
```
Comment (gertjanmaas) [inline modules/action-runner-binary-cache/lambdas/syncer/src/syncer/handler.ts:43]: Why not use the API? https://developer.github.com/v3/actions/self-hosted-runners/#list-runner-applications-for-a-repository

File: modules/action-runner-binary-cache/variables.tf [hunk 3]
```diff
@@ -19,3 +19,8 @@ variable "distribution_bucket_name" {
   type        = string
 }
 
+variable "lambda_schedule_expression" {
+  description = "Scheduler expression for action distributio syncer."
```
Comment (gertjanmaas) [inline modules/action-runner-binary-cache/variables.tf:23]:
```suggestion
  description = "Scheduler expression for action runner binary syncer."
```

File: modules/action-runner-binary-cache/lambdas/syncer/src/syncer/handler.ts (hunk 2 above)
Comment (npalm) [reply modules/action-runner-binary-cache/lambdas/syncer/src/syncer/handler.ts:43]: This API requires a repo based auth taken as far as I understand, tried to use a public API. We do not receive an app event so no option to use repo based token.

**Comment by npalm** [comment]
@gertjanmaas solved review comments, also add tagging for resources and update default example. Did not change the used API. The one you suggested is not a public api.

File: modules/action-runner-binary-cache/lambdas/syncer/src/syncer/handler.ts (hunk 2 above)
Comment (gertjanmaas) [reply modules/action-runner-binary-cache/lambdas/syncer/src/syncer/handler.ts:43]: It is also possible to use the org level api: https://developer.github.com/v3/actions/self-hosted-runners/#list-runner-applications-for-an-organization
If we use the Github App in that application we can generate tokens for both org and repo level if needed.

Comment (npalm) [reply modules/action-runner-binary-cache/lambdas/syncer/src/syncer/handler.ts:43]: I suggest we create an issue for this as improvement, the advantages of the current approach is that we do not require any auth.

**Pull request merged**